linera-execution/src/execution_state_actor.rs

```diff
@@ -267,10 +268,20 @@ where
                 callback.respond(bytes);
             }
 
-            FetchJson { url, callback } => {
-                let json: serde_json::Value = reqwest::get(url).await?.json().await?;
-                let string = serde_json::to_string_pretty(&json)?;
-                callback.respond(string);
+            FetchJson {
+                url,
+                payload,
+                callback,
+            } => {
+                let res = Client::new()
+                    .post(url)
+                    .body(payload)
+                    .header(CONTENT_TYPE, "application/json")
+                    .send()
+                    .await?;
+                let body = res.bytes().await?;
+                let bytes = body.as_ref().to_vec();
+                callback.respond(bytes);
```

Comment:
So now `fetch_json` is neither aware that it only _fetches_ data (it makes a `POST`!), nor that it has anything to do with JSON. That needs a better name.

Also, the doc comments on the runtime methods should warn that this only works if the body has a high probability of being byte-by-byte identical for every node making the request, since we now apply no normalization at all to it.

Reply:
What name would you suggest?
I can think of `post_payload`?

Reply:
Or `http_post` maybe?
(Since this is in the context of a contract runtime, a function to make HTTP posts might seem surprising at first, so we should be explicit.)

Reply:
Yes, I used `http_post`.

Reply:
Given that the content type is "application/json", it would make sense to have "json" in the name. Perhaps `CallJson[Rpc]` then?

Reply:
Ah, right, I forgot about the content type!

But now that we've generalized it and store the response bytes as the oracle data: Maybe the content type should just be another argument of `http_post`?

Reply:
Yes, let's ask another parameter for the content_type.